### PR TITLE
Added Prettier, ESLint & updated dependencies

### PR DESCRIPTION
**Pull request by Razican**
This commit updates all dependencies to latest available versions, adds ESLint, better typescript and ECMAScript module usage in config files, and adds Docusaurus 3.1.1 type checking.

I also fixed almost all lints. There is one left, but we need to check about that one separately, as we have a link without href, that responds to an on-click (so it could be seen as a button maybe).

We should at some point enforce "npm run lint" on every commit in the CI.

### CODE REVIEW

**Review by nekevss — APPROVED** [review]
TIL I need to lower my spaces to 2 😅

LGTM

**Comment by Razican** [comment]
> TIL I need to lower my spaces to 2 😅

It depends on the language it seems. It uses different defaults for different languages.

**Review by HalidOdat — APPROVED** [review]
Nice work! Looks good to me! :)

File: docusaurus.config.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        },
+        {
+          title: "Community",
+          items: [
+            {
+              label: "Discord",
+              href: "https://discord.gg/tUFFk9Ys",
+            },
+            {
+              label: "Mastadon",
```
Comment (jedel1043) [inline docusaurus.config.ts]:
```suggestion
              label: "Mastodon",
```

File: src/components/HomepageFeatures/index.tsx [hunk 2]
```diff
@@ -11,21 +15,21 @@ type FeatureItem = {
...
-    Svg: require("@site/static/img/new_logo_black.svg").default,
+    Svg: new_logo_black,
     description: (
       <>Boa hopes to bring Rust safety garuantees to the world of JS engines.</>
     ),
   },
   {
     title: "Aims for ECMAScript Conformance",
-    Svg: require("@site/static/img/new_logo_yellow.svg").default,
+    Svg: new_logo_yellow,
```
Comment (jasonwilliams) [inline src/components/HomepageFeatures/index.tsx:25]: Where do these SVGs appear? I don't see them on the site

**Review by jasonwilliams — APPROVED** [review]
Great work!

**Pull request merged**